geopandas/tools/tests/test_sjoin.py

```diff
@@ -183,7 +183,7 @@ def test_inner(self, predicate, predicate_kw, dfs):
             ].astype("int64")
         if index == "named-index":
             exp[["df1_ix", "df2_ix"]] = exp[["df1_ix", "df2_ix"]].astype("int64")
-            exp = exp.set_index("df1_ix").rename(columns={"df2_ix": "index_right"})
+            exp = exp.set_index("df1_ix")
```

Comment:
This is the original "bug" fix, that we preserve the name of the right index, if it had one, instead of always using the auto-generated "index_right"